### PR TITLE
Create Moore_voting_algorithm.dart

### PR DESCRIPTION
**Pull request by neha-hasija17**
This is an algorithm which is used to find the majority element in an array
This pull request is for HacktoberFest 2020

# Welcome to Dart community

### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: other/Moore_voting_algorithm.dart [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+void main() {
+  List<int> a1 = [1, 2, 2, 2, 2, 5, 1];
+  print(majorityElement(a1, a1.length));
+  List<int> a2 = [3, 3, 22, 21, 21, 5, 21];
+  print(majorityElement(a2, a2.length));
+  List<int> a3 = [30, 30, 40, 30, 40, 30, 40];
+  print(majorityElement(a3, a3.length));
+  List<int> a4 = [100, 4000, 220, 220, 220, 100, 4000];
+  print(majorityElement(a4, a4.length));
+  return;
```
Comment (Parowicz) [inline other/Moore_voting_algorithm.dart:36]: This `return` is redundant.
```suggestion
```

**Comment by neha-hasija17** [comment]
@Parowicz sir I am facing difficulty in adding the tests, as per your request, can you please help me out with the same?

File: other/Moore_voting_algorithm.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// Driver code
+void main() {
+  List<int> a1 = [1, 2, 2, 2, 2, 5, 1];
+  print(majorityElement(a1, a1.length));
+  List<int> a2 = [3, 3, 22, 21, 21, 5, 21];
+  print(majorityElement(a2, a2.length));
+  List<int> a3 = [30, 30, 40, 30, 40, 30, 40];
+  print(majorityElement(a3, a3.length));
+  List<int> a4 = [100, 4000, 220, 220, 220, 100, 4000];
+  print(majorityElement(a4, a4.length));
```
Comment (Parowicz) [inline other/Moore_voting_algorithm.dart:35]:
```suggestion
  test("majorityElement", () {
    List<int> a1 = [1, 2, 2, 2, 2, 5, 1];
    expect(majorityElement(a1, a1.length), equals(2));

    List<int> a2 = [30, 30, 40, 30, 40, 30, 40];
    expect(majorityElement(a2, a2.length), equals(30));
  });

  test("majorityElement returns -1 when there is no dominant element", () {
    List<int> a1 = [3, 3, 22, 21, 21, 5, 21];
    expect(majorityElement(a1, a1.length), equals(-1));

    List<int> a2 = [100, 4000, 220, 220, 220, 100, 4000];
    expect(majorityElement(a2, a2.length), equals(-1));
  });
```

File: other/Moore_voting_algorithm.dart [hunk 3]
```diff
@@ -0,0 +1,36 @@
+int majorityElement(List<int> arr, int n) {
```
Comment (Parowicz) [inline other/Moore_voting_algorithm.dart:1]:
```suggestion
import 'package:test/test.dart';

int majorityElement(List<int> arr, int n) {
```

**Comment by neha-hasija17** [comment]
@Parowicz please check

**Review by Parowicz — APPROVED** [review]
LGTM!

**Comment by neha-hasija17** [comment]
@Parowicz please remove the label of tests are needed from this PR. :)

**Pull request merged**